src/resource-archive/index.ts

```diff
@@ -42,8 +49,21 @@ class Watcher {
   }
 
   async idle() {
-    // TODO -- wait for network idle
-    await new Promise((r) => setTimeout(r, 1000));
+    // Wait for network idle, but wait a maximum of 4s.
+    const ac: AbortController = new AbortController();
+
+    // eslint-disable-next-line @typescript-eslint/no-implied-eval
+    let timeoutId: any;
+    const timeoutPromise = new Promise((r) => {
+      timeoutId = setTimeout(r, 4000);
```

Comment:
Should we add the unresolved requests/promises to the log?

Reply:
Yes, I think that's a good idea. I'll add logging information for this.